azurelinuxagent/common/protocol/metadata.py

```diff
@@ -394,13 +394,15 @@ def parse(self, json_text):
                 elif re.match(r'[-]+BEGIN.*CERTIFICATE[-]+', line):
                     begin_crt = True
                 elif re.match(r'[-]+END.*KEY[-]+', line):
+                    assert begin_prv
```

Comment:
asserting on begin_prv checks that you have previously seen a "BEGIN KEY" before seeing "END KEY" - improved defensive programming to handle possibly bad inputs

Reply:
I removed these checks entirely.  The openssl tools will fail if the certificate or private key data is invalid and this gets handled.  No need to look for a BEGIN at all since we're grabbing the entire content of the file anyway.